Prometheus Analysis SigV4 Support

# Summary

Prometheus analysis should support SigV4 when creating the prometheus client. This would allow it to query Amazon Managed Prometheus, which requires [SigV4](https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html) authentication.

The prometheus client already supports a [SigV4 RoundTripper](https://pkg.go.dev/github.com/prometheus/common/sigv4). This can be used when constructing the client to allow a user to authenticate with Amazon Managed Prometheus using the default GO credential provider chain.

# Use Cases

When performing prometheus analysis using Amazon Managed Prometheus as a metric data store.

---
<!-- Issue Author: Don't delete this message to encourage other users to support your issue! -->
**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritize the issues with the most 👍.

Hi @zachaller 

We tried to leverage this feature using the `latest` tag of the AR controller image but we were hit by an error:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x1d986c4]
goroutine 353 [running]:
github.com/prometheus/common/sigv4.NewSigV4RoundTripper(0x0, {0x0?, 0x0?})
	/go/pkg/mod/github.com/prometheus/common/sigv4@v0.1.0/sigv4.go:57 +0x64
github.com/argoproj/argo-rollouts/metricproviders/prometheus.NewPrometheusAPI({{0xc000fbbcc0, 0xc}, {0xc000fbbccc, 0x2}, {0x0, 0x0}, 0x0, {0xc000fbbcd0, 0x10}, {0xc000fbbce0, ...}, ...})
	/go/src/github.com/argoproj/argo-rollouts/metricproviders/prometheus/prometheus.go:176 +0x334
github.com/argoproj/argo-rollouts/metricproviders.(*ProviderFactory).NewProvider(_, {0xc0001f6000, 0xc000c2eb70, {0x0, 0x0, 0x0}, 0x0, 0x0, {0x0, 0x0}, ...}, ...)
	/go/src/github.com/argoproj/argo-rollouts/metricproviders/metricproviders.go:40 +0xe98
github.com/argoproj/argo-rollouts/analysis.(*Controller).runMeasurements.func1({{{0xc000fbbcc0, 0xc}, {0xc000fbbccc, 0x2}, {0x0, 0x0}, 0x0, {0xc000fbbcd0, 0x10}, {0xc000fbbce0, ...}, ...}, ...})
	/go/src/github.com/argoproj/argo-rollouts/analysis/analysis.go:328 +0x3fd
created by github.com/argoproj/argo-rollouts/analysis.(*Controller).runMeasurements
	/go/src/github.com/argoproj/argo-rollouts/analysis/analysis.go:319 +0x38a
```
We use the following code in our AnalysisTemplate:

```
apiVersion: argoproj.io/v1alpha1
kind: AnalysisTemplate
metadata:
  name: {{ include "damage-report-helm.backend.fullname" . }}-errors-rate
spec:
  args:
  - name: targetGroupName
  dryRun:
  - metricName: success-rate
  metrics:
  - name: success-rate
    interval: 5m
    successCondition: isNaN(result[0])
    failureCondition: result[0] >= 1
    failureLimit: 1
    provider:
     prometheus:
       address: https://aps-workspaces.eu-central-1.amazonaws.com/workspaces/ws-f77bee0c-0494-4267-b64c-91c938eb734b
       query: |
         sum(irate(http_requests_total{job="damage-report-backend",status="5xx"}[5m]))

```
I'm not sure if should I create another issue since release 1.5 is not out yet...

--- a/metricproviders/prometheus/prometheus.go
+++ b/metricproviders/prometheus/prometheus.go
@@ -171,9 +171,9 @@ func NewPrometheusAPI(metric v1alpha1.Metric) (v1.API, error) {
 	}
 	//Check if using Amazon Managed Prometheus if true build sigv4 client
 	if strings.Contains(metric.Provider.Prometheus.Address, "aps-workspaces") {
-		var cfg *sigv4.SigV4Config
+		var cfg sigv4.SigV4Config
 		var next http.RoundTripper
-		sigv4RoundTripper, err := sigv4.NewSigV4RoundTripper(cfg, next)
+		sigv4RoundTripper, err := sigv4.NewSigV4RoundTripper(&cfg, next)
 		if err != nil {
 			log.Errorf("Error creating SigV4 RoundTripper: %v", err)
 			return nil, err

